src/lib/core/assoc.h

```diff
@@ -189,6 +189,69 @@ mh_strnu32_find_str(struct mh_strnu32_t *h, const char *str, uint32_t len)
 	return mh_strnu32_find(h, &key, NULL);
 }
 
+/*
+ * Map: (const char *, uint32_t, const char *, uint32_t) => (void *)
+ */
+
+#define mh_name _strn_strn
+/**
+ * Key of `mh_strn_strn_node_t` hash table.
+ */
+struct mh_strn_strn_key_t {
```

Comment:
I don't like this new hash table type (primarily because of its name). I'd prefer to introduce a new hash table type exclusively for storing runtime Lua call privileges. However, if you're sure you want to add it, please:

1. Do it in a separate commit.
2. Name it consistently with other generic hash tables. I can't suggest a nice name but `mh_strn_strn` sounds like it's mapping `strn` to `strn`, not `strn+strn` to `ptr`. I guess, to be consistent with our other generic hash tables, it should be named `mh_strnstrnptr`.

Reply:
Renamed to `mh_strnstrnptr` and moved it to a separate commit.